### ISSUE TITLE
Investigate ERC4337: How much work to integrate? Is it worth doing?

### ISSUE THREAD

**Issue by JamesSmartCell** [AlphaWallet/alpha-wallet-android#2814]
EIP4337 is a neat and official way to make multi-sig wallets and social recovery wallets, and to reduce TX fees by bundling transactions.

### EIP4337: Account Abstraction

With EIP4337 there's a higher level mempool that we can submit transactions to that can, at some point get bundled up together and submitted as a single transaction to the main mempool. In addition it provides a way to fund transactions via third party - user submits the tx, contract pays for it under specific rules.

It seems like we could use this with attestations so people can mint NFT using their attestation without paying any gas (we have a contract to pay gas for this sole purpose). So this feature might be useful for attestation based NFT sites we build.

It also allows usage of different signature protocols, like BLS, which could reduce gas fees further and open up a new front of attestation use cases.

https://medium.com/infinitism/erc-4337-account-abstraction-without-ethereum-protocol-changes-d75c9d94dc4a
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4337.md

**Comment by hboon** [comment]
@JamesSmartCell There are a few account abstraction EIPs. I don't think any of them has been implemented yet.

**Comment by JamesSmartCell** [comment]
> @JamesSmartCell There are a few account abstraction EIPs. I don't think any of them has been implemented yet.

Yes, even though it was presented over a year ago it doesn't appear to have gone live. 

It looks like a useful standard for us, and looks like one of Vitalik's favourite improvements.

**Comment by hboon** [comment]
Yeah, IMO we will almost definitely hop on account abstraction at some point, just waiting to see which direction it goes. But definitely good to try it out and get a feel. Please write a "trip report".

> Vitalik's favourite

Well, smart contract wallets was all the hype and got all the support… :)
